dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/patch/operation/EPersonPasswordReplaceOperation.java

```diff
@@ -27,25 +37,59 @@
 @Component
 public class EPersonPasswordReplaceOperation<R> extends PatchOperation<R> {
 
+    private static final Logger log = org.apache.logging.log4j.LogManager
+        .getLogger(EPersonPasswordReplaceOperation.class);
+
     /**
      * Path in json body of patch that uses this operation
      */
     public static final String OPERATION_PASSWORD_CHANGE = "/password";
     protected EPersonService ePersonService = EPersonServiceFactory.getInstance().getEPersonService();
 
+    @Autowired
+    private RequestService requestService;
+
+    @Autowired
+    private AccountService accountService;
+
     @Override
     public R perform(Context context, R object, Operation operation) {
         checkOperationValue(operation.getValue());
         if (supports(object, operation)) {
             EPerson eperson = (EPerson) object;
+            String token = requestService.getCurrentRequest().getHttpServletRequest().getParameter("token");
             checkModelForExistingValue(eperson);
+            if (StringUtils.isNotBlank(token)) {
+                patchWithToken(context,eperson, token, operation);
+            }
+            if (!AuthorizeUtil.authorizeUpdatePassword(context, eperson.getEmail())) {
+                throw new DSpaceBadRequestException("Password cannot be updated for the given EPerson with email: " +
+                                                        eperson.getEmail());
+            }
             ePersonService.setPassword(eperson, (String) operation.getValue());
             return object;
         } else {
             throw new DSpaceBadRequestException("EPersonPasswordReplaceOperation does not support this operation");
         }
     }
 
+    private void patchWithToken(Context context, EPerson eperson, String token, Operation operation) {
```

Comment:
This method appears to be misnamed.  I don't see a patch going on here...it looks to just delete the token?

Reply:
This has been adjusted